ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -380,16 +388,24 @@ def _get_events_join_query(self) -> str:
             - funnel_step_names
         """
 
-        return f"""
+        aggregation_person_join = f"""
             JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) AS pdi
-                ON pdi.distinct_id = events.distinct_id
+                    ON pdi.distinct_id = events.distinct_id
+
+                -- NOTE: I would love to right join here, so we count get total
+                -- success/failure numbers in one pass, but this causes out of memory
+                -- error mentioning issues with right filling. I'm sure there's a way
+                -- to do it but lifes too short.
+                JOIN funnel_people AS person
+                    ON pdi.person_id = person.person_id
+        """
 
-            -- NOTE: I would love to right join here, so we count get total
-            -- success/failure numbers in one pass, but this causes out of memory
-            -- error mentioning issues with right filling. I'm sure there's a way
-            -- to do it but lifes too short.
+        aggregation_group_join = f"""
             JOIN funnel_people AS person
-                ON pdi.person_id = person.person_id
+                ON person.person_id = events.$group_{self._filter.aggregation_group_type_index}
+        """
+        return f"""
+            {aggregation_group_join if self._filter.aggregation_group_type_index is not None else aggregation_person_join}
```

Comment:
Nit, Suggestion: Pull out method for this? The variable assignments above are slightly complex and deserves a name.